### ISSUE TITLE
Anyone having luck getting redux-devtools-extension working?

### ISSUE THREAD

**Issue by leighghunt** [jaysoo/todomvc-redux-react-typescript#8]
Thank you for the tutorial and example code - I've been trying for a while to get the [redux-devtools-extension](https://github.com/zalmoxisus/redux-devtools-extension) working with this sample, but unsuccessfully.

I think it might be to do with the version of redux being included, but I think I'm having trouble getting the correct version of the typings to be referenced.

Issue - redux-devtools-extension is reporting:

_No store found. Make sure to follow the [instructions](https://github.com/zalmoxisus/redux-devtools-extension#implementation)._

When trying to follow those instructions, typescript is complaining about my call to `createStore` and returning the following error:

```
Supplied parameters do not match any signature of call target.
```

<img width="1314" alt="screen shot 2016-06-10 at 3 56 45 pm" src="https://cloud.githubusercontent.com/assets/349153/15954103/23ebcd8a-2f27-11e6-9092-dce9bf62b805.png">

I've tried updating versions in npm, but may be missing something necessary to get typings updated too? I think if I can get the [addition](https://github.com/DefinitelyTyped/DefinitelyTyped/commit/66189746241fd5014d3acbdeadf4b2b75b2772d9) to `createStore` it should be happy, but I can't work out the magical typings incantation.

Environment: Mac OSX, Atom with atom-typescript and atom-jsx-transform

**Comment by leighghunt** [comment]
For anyone else who runs into this issue, the resolution can be found at https://github.com/zalmoxisus/redux-devtools-extension/issues/134#issuecomment-225124613

``` javascript
const enhancer = window['devToolsExtension'] ? window['devToolsExtension']()(createStore) : createStore;
const store: IStore<any> = enhancer(rootReducer, initialState);
```

**Comment by smartin015** [comment]
I also had some issues getting redux-devtools-extension working in a custom environment with typescript (webpack + awesome-typescript-loader) and react-thunk middleware for async calls. For posterity, here's what worked for me:

``` typescript
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import rootReducer from '....'

let initialState = ....
let devtools: any = window['devToolsExtension'] ? window['devToolsExtension']() : (f:any)=>f;
let middleware = applyMiddleware(thunk);
const store: any = middleware(devtools(createStore))(rootReducer, initialState);
```

The "any"s are a bit sloppy, and I'll be refining them later. But it works!

**Comment by ali-habibzadeh** [comment]
[tslint] object access via string literals is disallowed (no-string-literal)

**Comment by godmar** [comment]
This worked for me (note the empty lines are significant)
```
import { applyMiddleware, createStore } from 'redux';

import rootReducer from './reducers/root';

import thunk from 'redux-thunk';

const w : any = window as any;
const devtools: any = w.devToolsExtension ? w.devToolsExtension() : (f:any)=>f;
const middleware = applyMiddleware(thunk);
const store: any = middleware(devtools(createStore))(rootReducer);

export default store;
```

**Comment by phiboi** [comment]
This is also a good bypass.

```typescript
const w: any = window;

export const store: Store = createStore(postReducer, compose(
  applyMiddleware(thunk),
  w.__REDUX_DEVTOOLS_EXTENSION__ && w.__REDUX_DEVTOOLS_EXTENSION__()
));
```

**Comment by phinguyen1618** [comment]
This is also a good bypass.

```typescript
export const store: Store = createStore(postReducer, compose(
  applyMiddleware(thunk),
  (window as any).__REDUX_DEVTOOLS_EXTENSION__ && (window as any).__REDUX_DEVTOOLS_EXTENSION__()
));
```

**Comment by mmakrzem** [comment]
I have redux v4 installed, but when I try to use : 

`_store = createStore( reducers, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__() `

I get this error in the console:

Uncaught TypeError: Cannot read property 'state' of undefined
    at unliftState (<anonymous>:2:31677)
    at Object.getState (<anonymous>:2:31745)
    at Function.getStoreState (AppStore.ts?a477:70)
    at Function.start (Renderer.ts?7ee2:68)
    at eval (index.ts?7fbc:5)
    at Object../src/renderer/index.ts (renderer.js:2589)
    at __webpack_require__ (renderer.js:680)
    at fn (renderer.js:90)
    at Object.0 (renderer.js:2605)
    at __webpack_require__ (renderer.js:680)

If instead I do:
`_store = createStore( reducers )`
then my app has no errors, but I don't have any redux devTools.

**Comment by Jalever** [comment]
> This is also a good bypass.
> 
> ```ts
> export const store: Store = createStore(postReducer, compose(
>   applyMiddleware(thunk),
>   (window as any).__REDUX_DEVTOOLS_EXTENSION__ && (window as any).__REDUX_DEVTOOLS_EXTENSION__()
> ));
> ```
thx, it works.

**Comment by thaxy** [comment]
You could also extend the Window interface (it automatically merges, all other props are still available):

```
declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION__?: () => StoreEnhancer;
  }
}

const store = createStore(
  rootReducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
);
```

Or you ``npm install --save redux-devtools-extension`` and then:

```
import { devToolsEnhancer } from "redux-devtools-extension";

const store = createStore(
  rootReducer,
  devToolsEnhancer({}),
);
```
